JAICore/jaicore-ml/src/jaicore/ml/core/dataset/attribute/multivalue/MultiValueAttributeType.java

```diff
@@ -0,0 +1,37 @@
+package jaicore.ml.core.dataset.attribute.multivalue;
+
+import java.util.Collection;
+import java.util.Set;
+
+/**
+ * The multi-value attribute type describes the domain a value of a respective multi-value attribute value stems from.
+ *
+ * @author wever
+ *
+ */
+public class MultiValueAttributeType implements IMultiValueAttributeType {
+
+	/** The domain defining where values of multi-value attributes stem from. */
+	private final Set<String> domain;
+
+	/**
+	 * Constructor setting the domain of the multi-value attribute values.
+	 *
+	 * @param domain
+	 *            The domain multi-value values may stem from.
+	 */
+	public MultiValueAttributeType(final Set<String> domain) {

```

Comment:
Isn't this very restrictive to say that we can have only strings in the domain of MultiValueAttributes?

Reply:
Yes, it is. It could be also some kind of binary or complex object. Ideas?

Reply:
One way would be to use generic types to give the user the option to set at least the domain. Other than that, not really, no.